sdk/identity/identity/src/credentials/managedIdentityCredential/imdsMsi.ts

```diff
@@ -47,6 +54,13 @@ function prepareRequestOptions(resource?: string, clientId?: string): RequestPre
   };
 }
 
+// 30s -> 60s -> 120s
+export const imdsMsiRetryConfig = {
```

Comment:
maybe dumb question, but why not break this up into 3 constant numbers, so we don't have to worry about them being changed? Technically, anyone who imports from this module can modify this object and affect all credential behavior thereby

Reply:
Two reasons: constants are harder to change or mock (for tests), and if we separated this, each constant would have a huge name. This way the `imdsMsiRetryConfig` can be locally renamed (like `config` or anything else) and the properties become easier to access.

Reply:
Sure, it just feels a little weird that a caller can alter a "constant" piece of default configuration. If this accidentally got exported outside the package, we'd technically be allowing for any caller to alter the behavior for other callers, which is a confusing contract.

If you wanted this to be more easily testable, it seems like passing in an optional config (with a default value) could accomplish this just as easily by binding the config to a closure that returns the `MSI` object

Reply:
e.g. `imdsMsi(config: ImdsMsiRetryConfig = defaultConfig): MSI`